### ISSUE TITLE
Step: Using Bind Mounts fails

### ISSUE THREAD

**Issue by hboukricha** [docker/getting-started#231]
When running the following docker command
```
docker run -dp 3000:3000 \
    -w /app -v "$(pwd):/app" \
    node:12-alpine \
    sh -c "yarn install && yarn run dev"
```
I get the following Error:

```
...
gyp ERR! configure error 
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (/app/node_modules/node-gyp/lib/configure.js:484:19)
gyp ERR! stack     at PythonFinder.<anonymous> (/app/node_modules/node-gyp/lib/configure.js:406:16)
gyp ERR! stack     at F (/app/node_modules/which/which.js:68:16)
gyp ERR! stack     at E (/app/node_modules/which/which.js:80:29)
gyp ERR! stack     at /app/node_modules/which/which.js:89:16
gyp ERR! stack     at /app/node_modules/isexe/index.js:42:5
gyp ERR! stack     at /app/node_modules/isexe/mode.js:8:5
gyp ERR! stack     at FSReqCallback.oncomplete (fs.js:168:21)
gyp ERR! System Linux 5.11.0-41-generic
gyp ERR! command "/usr/local/bin/node" "/app/node_modules/node-gyp/bin/node-gyp.js" "configure" "--fallback-to-build" "--module=/app/node_modules/sqlite3/lib/binding/napi-v3-linux-x64/node_sqlite3.node" "--module_name=node_sqlite3" "--module_path=/app/node_modules/sqlite3/lib/binding/napi-v3-linux-x64" "--napi_version=8" "--node_abi_napi=napi" "--napi_build_version=3" "--node_napi_label=napi-v3"
gyp ERR! cwd /app/node_modules/sqlite3
gyp ERR! node -v v12.22.7
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node /app/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --module=/app/node_modules/sqlite3/lib/binding/napi-v3-linux-x64/node_sqlite3.node --module_name=node_sqlite3 --module_path=/app/node_modules/sqlite3/lib/binding/napi-v3-linux-x64 --napi_version=8 --node_abi_napi=napi --napi_build_version=3 --node_napi_label=napi-v3' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/app/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:314:20)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1022:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
node-pre-gyp ERR! System Linux 5.11.0-41-generic
node-pre-gyp ERR! command "/usr/local/bin/node" "/app/node_modules/sqlite3/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /app/node_modules/sqlite3
node-pre-gyp ERR! node -v v12.22.7
node-pre-gyp ERR! node-pre-gyp -v v0.11.0
node-pre-gyp ERR! not ok 
Failed to execute '/usr/local/bin/node /app/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --module=/app/node_modules/sqlite3/lib/binding/napi-v3-linux-x64/node_sqlite3.node --module_name=node_sqlite3 --module_path=/app/node_modules/sqlite3/lib/binding/napi-v3-linux-x64 --napi_version=8 --node_abi_napi=napi --napi_build_version=3 --node_napi_label=napi-v3' (1)

```

I managed to solve this by simply using the formerly build docker image tagged as `getting-started`, which runs python etc. and changed the current command to:

```
docker run -dp 3000:3000 \
    -w /app -v "$(pwd):/app" \
    getting-started \
    sh -c "yarn install && yarn run dev"
```
I'm pretty new to docker, so I'm not sure about this, but could it be a valid option to use the formerly build image also for this tutorial step or is there another intention behind using the `node:12-alpine` image directly?

**Comment by marklackey** [comment]
I had the same issue, and I solved it similarly... I used the getting-started image, but I made a new one that I called getting-started-base, and I removed everything after the RUN line with the dependencies installs. I'm on an M1 mac. What platform are you on?

**Comment by hboukricha** [comment]
I'm on `Ubuntu 20.04`.

Regarding the container image, I would say it depends on the goal of the tutorial within this step to decide on how to define the steps within the dockerfile. For now, it is not clear to me why not including the `sh -c "yarn install && yarn run dev"` into the dockerfile and have for instance a new different container from `getting-started` one.

**Comment by Tobindekorne** [comment]
I had this same problem on M1 Mac and was able to solve by just adding to the command that we run when starting the container. I ran:
```
docker run -dp 3000:3000 \
    -w /app -v "$(pwd):/app" \
    node:12-alpine \
    sh -c "apk add --no-cache python2 g++ make && yarn install && yarn run dev"
```
and this worked as well without the need to create a new image.

**Comment by StefanScherer** [comment]
Thanks, yes we've just merged a PR that adds this command to the guide. It's happening on M1 machines or other ARM64 machines. Unfortunately on these machines the sqlite3 package needs the compiler and python installed to build the dependency.

**Comment by hboukricha** [comment]
@Tobindekorne I agree that there is no need to create a new image and putting the needed dependencies within the command is also working. However, it was not clear what is the tutorial's intention at this step, i.e. use an existing image, e.g., the getting-started one or put the dependencies into the command.

**Comment by ColemanDunn** [comment]
> Thanks, yes we've just merged a PR that adds this command to the guide. It's happening on M1 machines or other ARM64 machines. Unfortunately on these machines the sqlite3 package needs the compiler and python installed to build the dependency.

@StefanScherer After seeing your comment I can see that this command has been added to the guide on `localhost:80` after running `docker run -d -p 80:80 docker/getting-started` as per part 1 of the [online guide](https://docs.docker.com/get-started/) as shown here:
<img width="985" alt="image" src="https://user-images.githubusercontent.com/42652642/165375901-ab9ee23e-638d-498f-9e67-7142d3642b04.png">

But it has NOT been added to the equivalent page [for part 6 on the regular website](https://docs.docker.com/get-started/06_bind_mounts/).
<img width="1211" alt="image" src="https://user-images.githubusercontent.com/42652642/165376535-ecd8e4ab-4461-4b74-b174-bfc0d165aeaf.png">

Is there any possibility that this can be added to the regular website? It personally stopped me in my tracks and really hindered my progress. I came across many different pages before finally finding this solution and it could benefit others as well. Not to mention that Apple removed python 2 from the OS as of macOS Monterey 12.3 which was released in March so trying to debug this locally on a version newer than that won't get you anywhere. I am on a MacBook Pro M1 Max for more context.
